### PR TITLE
Crate Testing

### PR DESCRIPTION
**Pull request by hashmaster3k**
### This PRs additions:

Testing for crate API calls.

#### Queries
- Get all Crates
  - [x] Happy path testing
  - [x] Sad path testing
- Crate by ID
  - [x] Happy path testing
  - [x] Sad path testing

#### Mutations
- Create Crate
  - [x] Happy path testing
  - [x] Sad path testing
- Update Crate
  - [x] Happy path testing
  - [x] Sad path testing
- Remove Crate
  - [x] Happy path testing
  - [x] Sad path testing

### Notes:

Made a change to updateCrate resolver to return updated objects versus returning null. Also updated testing utils file to include additional needed properties for authentication check.

### CODE REVIEW

File: code/api/src/modules/tests/crate/mutation.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            done();
+        })
+
+        it('remove a crate', async (done) => {
+            const response = await request(server)
+                .post('/graphql')
+                .send({ query: `mutation { crateRemove(id:${id}) { name } }` })
+                .expect(200);
+            done();
+        })
```
Comment (D-Lessenden) [inline code/api/src/modules/tests/crate/mutation.test.js:47]: Should there be an expectation here? Like an expectation for the id that would return null or something along those lines?

File: code/api/src/modules/tests/crate/mutation.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        it('remove - missing id argument', async (done) => {
+            const response = await request(server)
+                .post('/graphql')
+                .send({ query: `mutation { crateRemove() { name } }` })
+                .expect(400);
+                console.log(response.body)
+            done();
+        })
```
Comment (D-Lessenden) [inline code/api/src/modules/tests/crate/mutation.test.js:74]: Great sad paths!

File: code/api/src/modules/tests/crate/query.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  describe('happy-paths', () => {
+    it('returns all crates', async (done) => {
+      const response = await request(server)
+        .post('/graphql')
+        .send({query: `{ crates(orderBy:"asc") { id name description createdAt updatedAt } }`})
+        .expect(200)
+          const results = response.body.data.crates
+          expect(results.length).toBe(6)
+          results.forEach(crate => {
```
Comment (D-Lessenden) [inline code/api/src/modules/tests/crate/query.test.js:18]: Great use of forEach here. Very thorough.

**Review by D-Lessenden — COMMENTED** [review]
If you squash your comments I'd be happy to merge.

File: code/api/src/modules/tests/crate/mutation.test.js (hunk 1 above)
Comment (hashmaster3k) [reply code/api/src/modules/tests/crate/mutation.test.js:47]: Yes...thanks for catching that.

**Edit: Fixed!**

**Pull request merged**